code-submissions-clustering-plugin/src/main/kotlin/org/jetbrains/research/code/submissions/clustering/cli/DistanceCalculationRunner.kt

```diff
@@ -0,0 +1,61 @@
+package org.jetbrains.research.code.submissions.clustering.cli
+
+import com.intellij.openapi.application.ApplicationStarter
+import com.intellij.openapi.diagnostic.Logger
+import com.xenomachina.argparser.ArgParser
+import org.jetbrains.research.code.submissions.clustering.load.context.builder.gumtree.GumTreeGraphContextBuilder
+import org.jetbrains.research.code.submissions.clustering.model.Language
+import org.jetbrains.research.code.submissions.clustering.util.*
+import java.io.File
+import java.nio.file.Paths
+import kotlin.system.exitProcess
+
+object DistanceCalculationRunner : ApplicationStarter {
+    private val logger = Logger.getInstance(this::class.java)
+    private var toBinary: Boolean = false
+    private lateinit var inputFilename: String
+    private lateinit var outputPath: String
+
+    override fun getCommandName(): String = "calculate-dist"
+
+    @Suppress("TooGenericExceptionCaught")
+    override fun main(args: MutableList<String>) {
+        try {
+            ArgParser(args.drop(1).toTypedArray()).parseInto(::TransformationsRunnerArgs).run {
+                inputFilename = Paths.get(input).toString()
+                outputPath = Paths.get(output).toString()
+                toBinary = serialize
+            }
+
+            val file = File(inputFilename)
+            val language = Language.PYTHON
+            val context = GumTreeGraphContextBuilder.getContext(language)
+            val submissionsGraph = file.toSubmissionsGraph().calculateDistances(context)
+
+            createFolder(outputPath)
```

Comment:
Also, this part with saving output is the same. You can create an abstract base class for runners to construct graphs and reuse the common functionality